db/system_keyspace.cc

```diff
@@ -819,7 +819,8 @@ schema_ptr system_keyspace::v3::truncated() {
         // regular columns
         {
                 {"position", int32_type},
-                {"segment_id", long_type}
+                {"segment_id", long_type},
+                {"domain", byte_type},
```

Comment:
I'm not sure if it's safe to add a column to the table like this...

note that we need to support downgrades in the middle of rolling upgrade, i.e. if not all nodes have upgraded to a new Scylla version yet, we still need to be able to stop the upgrade procedure by downgrading the nodes back to the older version

the nodes that have upgraded and then downgraded will be left with this column.

Don't know if it's a problem or not.

cc @tgrabiec

Reply:
Yes, such things must be documented too.

It's better to use a text field instead of a byte type to make it less fragile (commitlog_domain enum becomes an ABI).

Reply:
A hacky way to do this is to overload some segment_id bits to encode the domain. I can't say I like it.